internal/pkg/client/cache/dir.go

```diff
@@ -22,69 +22,251 @@ const (
+	Net string
+
+	// Shub provides the location of the Shub cache
+	Shub string
+}
+
+// Create a new Singularity cache
+func Create() (*SingularityCache, error) {
+	// Singularity makes the following assumptions:
+	// - the default location for caches is specified by RootDefault
+	// - a user can specify the environment variable specified by DirEnv to change the location
+	// - a user can change the location of a cache at any time
+	// - but in the context of a Singularity command, the cache location cannot change once the command starts executing
+	basedir, err := getCacheBasedir()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get root of the cache: %s", err)
+	}
+
+	return Init(basedir)
+}
+
+// Init initializes a new cache within a given directory
+func Init(baseDir string) (*SingularityCache, error) {
+	rootDir := getCacheRoot(baseDir)
+	if err := initCacheDir(rootDir); err != nil {
+		return nil, fmt.Errorf("failed initializing caching directory: %s", err)
+	}
+
+	newCache := new(SingularityCache)
+	savedDirEnv := os.Getenv(DirEnv)
+	os.Setenv(DirEnv, baseDir)
+	newCache.PreviousDirEnv = savedDirEnv
+
+	newCache.BaseDir = baseDir
+	isDefaultCache, err := isDefaultBasedir(baseDir)
+	if err != nil {
+		return nil, fmt.Errorf("failed to check if this is the default cache: %s", err)
+	}
+	newCache.Root = rootDir
+	newCache.State = StateInitialized
+	newCache.Default = isDefaultCache
+	newCache.Library, err = getLibraryCachePath(newCache)
+	if err != nil {
+		return nil, fmt.Errorf("failed getting the path to the Library cache: %s", err)
+	}
+	newCache.OciTemp, err = getOciTempCachePath(newCache)
+	if err != nil {
+		return nil, fmt.Errorf("failed getting the path to the OCI temp cache")
+	}
+	newCache.OciBlob, err = getOciBlobCachePath(newCache)
+	if err != nil {
+		return nil, fmt.Errorf("failed getting the path to the OCI blob cache")
+	}
+	newCache.Net, err = getNetCachePath(newCache)
+	if err != nil {
+		return nil, fmt.Errorf("failed getting the path to the Net cache")
+	}
+	newCache.Shub, err = getShubCachePath(newCache)
+	if err != nil {
+		return nil, fmt.Errorf("failed getting the path to the Shub cache")
+	}
 
-// Root is the root location where all of singularity caching happens. Library, Shub,
-// and oci image formats supported by containers/image repository will be cached inside
-//
-// Defaults to ${HOME}/.singularity/cache
-func Root() string {
-	updateCacheRoot()
+	// Sanity check to ensure that everything is fine before returning the
+	// handle. We do not know a way at the moment to reach the error case.
+	if !newCache.IsValid() {
+		return nil, fmt.Errorf("unable to correctly initialize new cache")
+	}
 
-	return root
+	return newCache, nil
 }
 
-// Clean : wipes all files in the cache directory, will return a error if one occurs
-func Clean() error {
-	sylog.Debugf("Removing: %v", Root())
+// Destroy a specific Singularity cache
+func (c *SingularityCache) Destroy() error {
```

Comment:
in the same sense, this _is_ destroying the cache (L#155) but comments suggest more than one cache could be created at the same time.

Reply:
Good point, I will fix it.

Reply:
fixed